pkg/volume/volume.go

```diff
@@ -159,19 +159,27 @@ func (v *VolumeOperation) Delete() (*v1alpha1.CASVolume, error) {
 	if len(v.volume.Name) == 0 {
 		return nil, fmt.Errorf("unable to delete volume: volume name not provided")
 	}
+	// fetch the pv specifications
+	pv, err := v.k8sClient.GetPV(v.volume.Name, mach_apis_meta_v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
 
-	// TODO
-	// Get the PV details & extract the SC & then SC details
-	//  Get the CAS Template name for delete
-	//
-	// cas template to delete a cas volume
-	castName := v.volume.Annotations[string(v1alpha1.CASTemplateKeyForVolumeDelete)]
+	// get the storage class name corresponding to this volume
+	scName := pv.Spec.StorageClassName
+	if len(scName) == 0 {
+		return nil, fmt.Errorf("unable to delete volume %s: missing storage class in PV object", v.volume.Name)
+	}
+
+	// fetch the storage class specifications
+	sc, err := v.k8sClient.GetStorageV1SC(scName, mach_apis_meta_v1.GetOptions{})
+	if err != nil {
+		return nil, err
```

Comment:
Is the error logged inside caller? If not, it needs to be logged here.

Reply:
All the errors gets logged inside caller